src/bootstrap/compile.rs

```diff
@@ -598,6 +573,166 @@ impl Step for RustcLink {
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct CodegenBackend {
+    pub compiler: Compiler,
+    pub target: Interned<String>,
+    pub backend: Interned<String>,
+}
+
+impl Step for CodegenBackend {
+    type Output = ();
+    const ONLY_HOSTS: bool = true;
+    const DEFAULT: bool = true;
+
+    fn should_run(run: ShouldRun) -> ShouldRun {
+        run.path("src/librustc_trans").krate("rustc_trans")
```

Comment:
`krate` here is useless in theory since that crate doesn't have a crate graph really in `src/` AFAIK. Though I guess it doesn't hurt... I'd prefer that we didn't have it though.